Fixed ability to tab-complete on $error[0]

SteveL-MSFT opened the pull request:
$error is actually $global:error, however, internally it's updated as ExecutionContext.DollarErrorVariable even though an `error` variable is in the variable table of the session state (however it's not writable).  During CommandCompletion this `error` variable is found before getting to global scope, but is empty so tab-complete returns no results.  Fix is to transform CommandCompletion calls to $error as $global:error so that it is found in the correct global scope.

Also fixed some hardcoded strings to use already defined consts for variable names related to Error.

Addresses https://github.com/PowerShell/PowerShell/issues/3373

iSazonov commented on src/System.Management.Automation/engine/runtime/Operations/VariableOps.cs:
```diff
@@ -170,6 +170,10 @@ internal static object GetVariableValue(VariablePath variablePath, ExecutionCont
             SessionStateScope scope;
+            if (String.Compare(variablePath.QualifiedName, "error", StringComparison.OrdinalIgnoreCase) == 0)
+            {
+                variablePath = new VariablePath("global:error");
```
Can we use defined constants here too?

SteveL-MSFT replied:
Yes, embarrassed I did that :)

iSazonov replied:
Closed.

iSazonov commented on test/powershell/Host/TabCompletion/BugFix.Tests.ps1:
```diff
@@ -54,4 +54,15 @@ Describe "Tab completion bug fix" -Tags "CI" {
+        $cmd = "5/0;`$error[0]."
+        $result = TabExpansion2 -inputScript $cmd -cursorColumn $cmd.Length
+        $result.CompletionMatches.Count | Should Be ($e | Get-Member).Count
+    }
```
All tests above checks `CompletionText`. Maybe add that for `$cmd = "5/0;`$error[0].FullyQ` ?

SteveL-MSFT replied:
Will do.

iSazonov replied:
Closed.

SteveL-MSFT commented:
AppVeyor failure is due to regression in ExecutionPolicy tests.  Investigating.

SteveL-MSFT committed:
Fixed ability to tab-complete on $error[0]

$error is actually $global:error, however, internally it's updated as ExecutionContext.DollarErrorVariable even though an error variable is in the variable table of the session state (however it's not writable). During CommandCompletion this error variable is found before getting to global scope, but is empty so tab-complete returns no results. Fix is to transform CommandCompletion calls to $error as $global:error so that it is found in the correct global scope.

Also fixed some hardcoded strings to use already defined consts for variable names related to Error.

iSazonov commented on test/powershell/Host/TabCompletion/BugFix.Tests.ps1:
```diff
@@ -54,4 +54,26 @@ Describe "Tab completion bug fix" -Tags "CI" {
+        }
+        catch {
+            $e = $error[0]
+        }
```
The try-catch is not used and can be removed.

SteveL-MSFT replied:
yup, cut and paste error

SteveL-MSFT commented:
The original fix I made had a bad assumption as there is a valid $script:error variable.  Coming up with a new solution.

SteveL-MSFT commented:
Closing this until I come up with a better solution
